src/test/java/TypewiseAlert/TypewiseAlertTest.java

```diff
@@ -1,15 +1,127 @@
+        assertTrue(CoolingType.MED_ACTIVE_COOLING.getLowTemp() == 0);
+        assertTrue(CoolingType.MED_ACTIVE_COOLING.getHighTemp() == 40);
+        assertTrue(CoolingType.HI_ACTIVE_COOLING.getLowTemp() == 0);
+        assertTrue(CoolingType.HI_ACTIVE_COOLING.getHighTemp() == 45);
+    }
+
+    @Test
+    public void sendToEmailTargetTest() {
+        String responseMsg = emailDispatcher.dispatchBreachMsgToTarget(BreachType.TOO_LOW);
+        assertNotNull(responseMsg);
+    }
+
+    @Test
+    public void sendToTargetControllerTest() {
+        String controllerMsg = deviceController.dispatchBreachMsgToTarget(BreachType.TOO_HIGH);
+        assertNotNull(controllerMsg);
+    }
+
+    @Test
+    public void testControllerResponseForLowTemp() {
+        BreachType breachType = BreachType.TOO_LOW;
+        int header = 0xfeed;
+        String controllerMsg = deviceController.dispatchBreachMsgToTarget(breachType);
+        String expectedMsg = String.format("%s : %s",header,breachType.getBreachMsg());
+        assertEquals(controllerMsg,expectedMsg);
+    }
+
+    @Test
+    public void testControllerResponseForHighTemp() {
+        BreachType breachType = BreachType.TOO_HIGH;
+        int header = 0xfeed;
+        String controllerMsg = deviceController.dispatchBreachMsgToTarget(breachType);
+        String expectedMsg = String.format("%s : %s",header,breachType.getBreachMsg());
+        assertEquals(controllerMsg,expectedMsg);
+    }
+
+    @Test
+    public void testNormalTempMsgDispatch() {
+        String controllerMsg = deviceController.dispatchBreachMsgToTarget(BreachType.NORMAL);
+        assertNull(controllerMsg);
+        String emailMsg = emailDispatcher.dispatchBreachMsgToTarget(BreachType.NORMAL);
+        assertNull(emailMsg);
+    }
+
+
+    @Test
+    public void testHighTempEmailDispatchMessages() {
+       String recipient =  "a.b@c.com";
+       BreachType highTempBreach = BreachType.TOO_HIGH;
+       String actualMsg =  emailDispatcher.dispatchBreachMsgToTarget(highTempBreach);
+       String expectedContent = String.format("To: %s\n %s",recipient,highTempBreach.getBreachMsg());
+       assertEquals(actualMsg,expectedContent);
+    }
+
+    @Test
+    public void testLowTempEmailDispatchMessages() {
+        String recipient =  "a.b@c.com";
+        BreachType lowTempBreach = BreachType.TOO_LOW;
+        String actualMsg =  emailDispatcher.dispatchBreachMsgToTarget(lowTempBreach);
+        String expectedContent = String.format("To: %s\n %s",recipient,lowTempBreach.getBreachMsg());
+        System.out.println(expectedContent);
+        assertEquals(actualMsg,expectedContent);
+    }
+
+    @Test
+    public void testAlertDispatcherFactory() {
+
+        IAlertTarget iAlertTarget = AlertDispatcherFactory.getRelevantDispatcherTarget(AlertTarget.TO_EMAIL);
+        assertNotNull(iAlertTarget);
+        iAlertTarget = AlertDispatcherFactory.getRelevantDispatcherTarget(AlertTarget.TO_EMAIL);
+        assertNotNull(iAlertTarget);
```

Comment:
how do we verify that the alert target is of type email? the test case needs to be strengthened here...